bioimageio/spec/__main__.py

```diff
@@ -1,96 +1,64 @@
+def package(rdf_source: str, auto_convert: bool = False):
+    raise NotImplementedError
 
 
 @app.command()
-def verify_spec(model_yaml: Path, auto_convert: bool = False):
-    try:
-        model_data = yaml.load(model_yaml)
-    except Exception as e:
-        pprint(e)
-        code = 1
-    else:
-        try:
-            if auto_convert:
-                model_data = maybe_convert_model(deepcopy(model_data))
-
-            verify_model_data(model_data)
-        except ValidationError as e:
-            pprint(e.messages)
-            code = 1
-        else:
-            code = 0
-            print(f"successfully verified model {model_yaml}")
+def verify(rdf_source: str, auto_convert: bool = False, auto_convert_inner: bool = None):
+    if auto_convert_inner is None:
+        auto_convert_inner = auto_convert
 
-    raise typer.Exit(code=code)
+    raise typer.Exit(code=_verify(rdf_source, auto_convert, auto_convert_inner))
 
 
-def verify_model_data(model_data: dict):
-    schema.Model().load(model_data)
-
-
-def verify_bioimageio_manifest_data(manifest_data: dict, auto_convert: bool = False):
+def _verify(rdf_source: str, auto_convert: bool, auto_convert_inner: bool) -> int:
     try:
-        if auto_convert:
-            manifest_data = maybe_convert_manifest(deepcopy(manifest_data))
-
-        manifest = schema.BioImageIoManifest().load(manifest_data)
-    except ValidationError as e:
-        pprint(e.messages)
+        raw_node = load_raw_node(rdf_source, update_to_current_format=auto_convert)
+    except Exception as e:
+        print(f"Could not verify {rdf_source}:")
+        pprint(e)
         return 1
+    else:
+        ret = 0
+        print(f"successfully verified {raw_node.type} {rdf_source}")
+        if raw_node.type == "collection":
+            for inner_category in ["application", "collection", "dataset", "model", "notebook"]:
+                for inner in getattr(raw_node, inner_category) or []:
+                    try:
+                        inner_source = inner.source
+                    except Exception as e:
+                        pprint(e)
+                        ret += 1
+                    else:
+                        ret += _verify(inner_source, auto_convert_inner, auto_convert_inner)
+
+        return ret
 
-    code = 0
-    for model in manifest["model"]:
-        try:
-            response = requests.get(model["source"], stream=True)
-            model_data = yaml.load(response.content)
-            if auto_convert:
-                model_data = maybe_convert_model(model_data)
-
-            verify_model_data(model_data)
-        except ValidationError as e:
-            print("invalid model:", model["source"])
-            pprint(e.messages)
-            code = 1
-        except Exception as e:
-            print("invalid model:", model["source"])
-            pprint(e)
-            code = 1
 
-    return code
+@app.command()
+def verify_spec(model_yaml: str, auto_convert: bool = False):
```

Comment:
Can we add some doc strings / help for this? I.e. `verify_spec --help` should print something useful.